os_net_config/impl_nmstate.py

```diff
@@ -850,9 +867,11 @@ def generate_rules(self):
                 break
         if clear_rules:
             for c_rule in curr_rules:
-                c_rule[NMRouteRule.STATE] = NMRouteRule.STATE_ABSENT
-                del_rules.append(c_rule)
-                logger.info(f'Prepare to remove rule - {c_rule}')
+                if self.managed_iprules(c_rule):
+                    c_rule[NMRouteRule.STATE] = NMRouteRule.STATE_ABSENT
+                    del_rules.append(c_rule)
+                    logger.info(f'Prepare to remove rule - {c_rule}')
+        self.write_config(MNG_RULES, add_rules)
```

Comment:
Keep a copy of the file content and write only after apply is successful. In case of failures the initial content shall be updated in the file.

Reply:
Also the rules shall be written in nmstate template format.